packages/ember-glimmer/tests/integration/components/attrs-lookup-test.js

```diff
@@ -0,0 +1,107 @@
+import { RenderingTest, moduleFor } from '../../utils/test-case';
+import { Component } from '../../utils/helpers';
+import { set } from 'ember-metal/property_set';
+
+moduleFor('Components test: attrs lookup', class extends RenderingTest {
+  ['@test it should be able to lookup attrs without `attrs.` - template access']() {
+    this.registerComponent('foo-bar', { template: '{{first}}' });
+
+    this.render(`{{foo-bar first=firstAttr}}`, {
+      firstAttr: 'first attr'
+    });
+
+    this.assertText('first attr');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('first attr');
+
+    this.runTask(() => set(this.context, 'firstAttr', 'second attr'));
+
+    this.assertText('second attr');
+
+    this.runTask(() => set(this.context, 'firstAttr', 'first attr'));
+
+    this.assertText('first attr');
+  }
+
+  ['@test it should be able to lookup attrs without `attrs.` - component access'](assert) {
+    let instance;
+
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        instance = this;
+      }
+    });
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{first}}' });
+
+    this.render(`{{foo-bar first=firstAttr}}`, {
+      firstAttr: 'first attr'
+    });
+
+    assert.equal(instance.get('first'), 'first attr');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(instance.get('first'), 'first attr');
+
+    this.runTask(() => set(this.context, 'firstAttr', 'second attr'));
+
+    assert.equal(instance.get('first'), 'second attr');
+
+    this.runTask(() => set(this.context, 'firstAttr', 'first attr'));
+
+    this.assertText('first attr');
+  }
+
+  ['@htmlbars should be able to modify a provided attr into local state #11571 / #11559'](assert) {
+    let instance;
+
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        instance = this;
+      },
+
+      didReceiveAttrs() {
+        this.set('first', this.attrs.first.toUpperCase());
+      }
+    });
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{first}}' });
+
+    this.render(`{{foo-bar first="first attr"}}`);
```

Comment:
@chancancode My understanding is that once you set the local state, it no longer is a CP (or it somehow breaks the shadowing link) so i can't do the U part of the INUR. I actually tried this and it failed to change the value; the existing test doesn't actually update the value of the dynamic arg so this was never tested.